Añadir tablas de DEpartment for Transportation (UK) para informacion en puertos:

In [27]:
import pandas as pd
import xlrd
import re

**Port0602**: counts of vessels - cargo and passenger - arriving at UK major ports, and selected other ports, from 2009 

In [96]:
ficheros_uk = '../home_uk/port0602.xls' # Otras fuentes: 
uk_number_vessels_ports = pd.read_excel(ficheros_uk,
                                   sheet_name ='port0602',
                                   header=8,
                                   usecols="A:Q",
                                   skiprows=[9,122,123,124,125,126,127,128,129],
                                   engine='xlrd'
                                   )

In [97]:
uk_number_vessels_ports

Port Major/Minor ports                Port group  2009  \
0               Aberdeen        Major port       Scotland East Coast  1490   
1              Appledore        Minor port           Bristol Channel     0   
2             Ardrishaig        Minor port       Scotland West Coast    62   
3                    Ayr        Minor port       Scotland West Coast   166   
4                 Barrow        Minor port         Lancs and Cumbria    39   
..                   ...               ...                       ...   ...   
107  Weymouth & Portland        Minor port              West Country   402   
108           Whitstable        Minor port           Thames and Kent    22   
109                 Wick        Minor port       Scotland East Coast     0   
110              Wisbech        Minor port  Wash & Northern E Anglia    23   
111           Workington        Minor port         Lancs and Cumbria    60   

     2010  2011  2012  2013  2014  2015  2016  2017  2017.1  2018  2019  2020  \
0    1525  1558  1536  1511  1488  1411  1344  1251    1266  1298  1145   886   
1       0     0     0     0     1     1     0     0       0     0     0     0   
2      81   110    65    89    87    85    75    19      15    11    15     5   
3     244   160   125   173   148   134   120    84     165   197   247   237   
4      74    67    52    59    40    43    47    51      61    54    59    52   
..    ...   ...   ...   ...   ...   ...   ...   ...     ...   ...   ...   ...   
107   361   953   827   721   658   172   394   371     491   508   581   375   
108    44    40    30    30    27    25    30    36       2     0    14     4   
109     1     0     0     0     7    20     5     5      13    12    24    16   
110    25    30    38    16    35    43    44    57      63    90    96    59   
111    50    91   132   146   152   143   123   136     111   177   159   138   

     2021  
0     939  
1       0  
2      13  
3     264  
4      51  
..    ...  
107   381  
108     9  
109     7  
110   103  
111   112  

[112 rows x 17 columns]

Entre 2017 y 2017.1 hay un cambio en la contabilidad

**Port0601**: counts of vessels - cargo and passenger - arriving at UK major ports, and selected other ports, from 2009 **by vessel type and size**\. $\red{Rechazada~por~falta~de~granularidad}$
**Port0302**: UK major port freight traffic by port and route (filter by direction, cargo group and year):
    - $\orange{Aporta~informacion~sobre~tipo~de~barco~entrante~o~saliente~con~cargamento~y~puerto.~PUEDE~SERVIR}$\
    - port0302: Table with Port, freight by scope\
    - Data: Year | Port | Cargo Group | REgion | Direction | Tonnage | Units



**Port0301**: UK major port freight traffic by port and year (filter by direction and cargo type), from 2000\
Metric: Tonnage

In [98]:
ficheros_uk = '../home_uk/port0301.xls' # Otras fuentes: 
uk_ports_freight = pd.read_excel(ficheros_uk,
                                   sheet_name ='port0301', # ver hoja 'Data'
                                   header=10,
                                   usecols="A:Q",
                                   skiprows=[40,46,59,67,68],
                                   engine='xlrd'
                                   )

In [99]:
uk_ports_freight

UK Country                         Port       2000       2001  \
0            England                       Boston   1264.756    847.184   
1            England                      Bristol   9646.908  10895.045   
2            England                        Dover  17433.742  19073.725   
3            England                   Felixstowe  29686.079  28353.531   
4            England           Fleetwood [Note 1]   1529.861   1607.977   
5            England                        Fowey   1527.476   1535.342   
6            England                        Goole   2710.591   2633.452   
7            England               Great Yarmouth    756.858    666.069   
8            England          Grimsby & Immingham  52500.648  54830.711   
9            England                      Harwich   3989.698   2622.573   
10           England                      Heysham   3723.492   3823.501   
11           England                         Hull  10722.469  10585.613   
12           England             Ipswich [Note 2]   2925.389   2923.963   
13           England                    Liverpool  30420.766  30288.314   
14           England              London [Note 5]  47891.966  50654.008   
15           England                   Manchester   7686.681   7879.260   
16           England                       Medway  15292.143  14852.590   
17           England                     Newhaven    578.375    997.600   
18           England                     Plymouth   1799.072   1877.374   
19           England                        Poole   1296.486   1819.141   
20           England                   Portsmouth   4521.104   4282.034   
21           England                     Ramsgate   1237.238   1431.580   
22           England                  River Trent   2449.503   2395.789   
23           England         Rivers Hull & Humber   9014.624   7845.541   
24           England                     Shoreham   1762.330   1803.946   
25           England         Southampton [Note 4]  34773.043  35689.032   
26           England                   Sunderland    934.410   1021.455   
27           England            Tees & Hartlepool  51472.639  50841.608   
28           England                         Tyne   2391.008   2468.886   
29  Northern Ireland                      Belfast  12483.808  13402.468   
30  Northern Ireland  Kilroot Power Station Jetty   1196.395   1264.288   
31  Northern Ireland                        Larne   4507.610   3519.739   
32  Northern Ireland                  Londonderry   1133.129   1059.926   
33  Northern Ireland                  Warrenpoint   1675.721   1480.397   
34          Scotland                     Aberdeen   3377.419   3845.418   
35          Scotland                    Cairnryan   2282.765   2014.228   
36          Scotland                        Clyde   7223.647  11068.826   
37          Scotland               Cromarty Firth   2329.108   2144.692   
38          Scotland                       Dundee   1046.582   1100.906   
39          Scotland                        Forth  41142.631  41607.233   
40          Scotland                    Glensanda   5898.967   5470.854   
41          Scotland           Loch Ryan [Note 3]      0.000      0.000   
42          Scotland                       Orkney  22797.699  18406.739   
43          Scotland                    Peterhead   1123.049   1339.258   
44          Scotland           Stranraer [Note 3]   1505.937   1404.379   
45          Scotland                   Sullom Voe  38204.065  31165.651   
46             Wales                      Cardiff   2699.400   2739.000   
47             Wales                    Fishguard    421.159    341.396   
48             Wales                     Holyhead   3444.453   3228.607   
49             Wales                Milford Haven  33767.544  33791.592   
50             Wales                      Newport   2672.696   2980.209   
51             Wales                  Port Talbot  11724.801   8271.000   
52             Wales               

**port0203**: UK major port unitised main freight traffic by port, cargo type, metric and loaded status (filter by direction and year)\
**fle0100**: UK Ship Register annual statistics and commercial data comparison: 2003 to 2023 (No desagregado)

Incluir los ficheros extraidos del zip:
- **port0502**: UK major port traffic by port (filter by direction), quarterly from 2009 (metric: thousand tonnes (tonnage). $\green{Puede~servir}$)
- **port0503**: UK major port traffic by port (filter by direction and cargo group). $\orange{No~se~perciben~muchas~diferencias~respecto~a~port0502}$

### Union de dataframes con Puertos:

1. Limpiar columnas con [Notes]

In [100]:
def remove_note(text):
    return re.sub(r"\[Note \d+\]", "", text)

In [101]:
uk_ports_freight['Port'] = uk_ports_freight['Port'].apply(remove_note)

In [102]:
uk_ports_freight

UK Country                         Port       2000       2001  \
0            England                       Boston   1264.756    847.184   
1            England                      Bristol   9646.908  10895.045   
2            England                        Dover  17433.742  19073.725   
3            England                   Felixstowe  29686.079  28353.531   
4            England                   Fleetwood    1529.861   1607.977   
5            England                        Fowey   1527.476   1535.342   
6            England                        Goole   2710.591   2633.452   
7            England               Great Yarmouth    756.858    666.069   
8            England          Grimsby & Immingham  52500.648  54830.711   
9            England                      Harwich   3989.698   2622.573   
10           England                      Heysham   3723.492   3823.501   
11           England                         Hull  10722.469  10585.613   
12           England                     Ipswich    2925.389   2923.963   
13           England                    Liverpool  30420.766  30288.314   
14           England                      London   47891.966  50654.008   
15           England                   Manchester   7686.681   7879.260   
16           England                       Medway  15292.143  14852.590   
17           England                     Newhaven    578.375    997.600   
18           England                     Plymouth   1799.072   1877.374   
19           England                        Poole   1296.486   1819.141   
20           England                   Portsmouth   4521.104   4282.034   
21           England                     Ramsgate   1237.238   1431.580   
22           England                  River Trent   2449.503   2395.789   
23           England         Rivers Hull & Humber   9014.624   7845.541   
24           England                     Shoreham   1762.330   1803.946   
25           England                 Southampton   34773.043  35689.032   
26           England                   Sunderland    934.410   1021.455   
27           England            Tees & Hartlepool  51472.639  50841.608   
28           England                         Tyne   2391.008   2468.886   
29  Northern Ireland                      Belfast  12483.808  13402.468   
30  Northern Ireland  Kilroot Power Station Jetty   1196.395   1264.288   
31  Northern Ireland                        Larne   4507.610   3519.739   
32  Northern Ireland                  Londonderry   1133.129   1059.926   
33  Northern Ireland                  Warrenpoint   1675.721   1480.397   
34          Scotland                     Aberdeen   3377.419   3845.418   
35          Scotland                    Cairnryan   2282.765   2014.228   
36          Scotland                        Clyde   7223.647  11068.826   
37          Scotland               Cromarty Firth   2329.108   2144.692   
38          Scotland                       Dundee   1046.582   1100.906   
39          Scotland                        Forth  41142.631  41607.233   
40          Scotland                    Glensanda   5898.967   5470.854   
41          Scotland                   Loch Ryan       0.000      0.000   
42          Scotland                       Orkney  22797.699  18406.739   
43          Scotland                    Peterhead   1123.049   1339.258   
44          Scotland                   Stranraer    1505.937   1404.379   
45          Scotland                   Sullom Voe  38204.065  31165.651   
46             Wales                      Cardiff   2699.400   2739.000   
47             Wales                    Fishguard    421.159    341.396   
48             Wales                     Holyhead   3444.453   3228.607   
49             Wales                Milford Haven  33767.544  33791.592   
50             Wales                      Newport   2672.696   2980.209   
51             Wales                  Port Talbot  11724.801   8271.000   
52             Wales               

In [103]:
uk_number_vessels_ports.columns

Index([            ' Port', 'Major/Minor ports',        'Port group',
                      2009,                2010,                2011,
                      2012,                2013,                2014,
                      2015,                2016,                2017,
                  '2017.1',                2018,                2019,
                      2020,                2021],
      dtype='object')

In [104]:
uk_number_vessels_ports[' Port'] = uk_number_vessels_ports[' Port'].apply(remove_note)

In [105]:
uk_number_vessels_ports.rename(columns={' Port':'Port'}, inplace=True)

In [106]:
uk_number_vessels_ports.columns

Index([             'Port', 'Major/Minor ports',        'Port group',
                      2009,                2010,                2011,
                      2012,                2013,                2014,
                      2015,                2016,                2017,
                  '2017.1',                2018,                2019,
                      2020,                2021],
      dtype='object')

In [107]:
uk_ports_freight.columns

Index(['UK Country', 'Port', '2000', '2001', '2002', '2003', '2004', '2005',
       '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013', '2014'],
      dtype='object')

In [108]:
df1 = uk_number_vessels_ports[["Port","Major/Minor ports","Port group"]]
df2 = uk_ports_freight[["Port","UK Country"]]

In [109]:
df_ports_uk = pd.merge(left = df1,
         right = df2,
         how = 'outer',
         left_on=['Port', df1.groupby('Port').cumcount()],
         right_on=['Port', df2.groupby('Port').cumcount()])

In [110]:
df_ports_uk['id_Port'] = df_ports_uk.index

In [111]:
df_ports_uk

Port  key_1 Major/Minor ports           Port group  \
0                Aberdeen      0        Major port  Scotland East Coast   
1               Appledore      0        Minor port      Bristol Channel   
2              Ardrishaig      0        Minor port  Scotland West Coast   
3                     Ayr      0        Minor port  Scotland West Coast   
4                  Barrow      0        Minor port    Lancs and Cumbria   
..                    ...    ...               ...                  ...   
114               London       0               NaN                  NaN   
115            Portsmouth      0               NaN                  NaN   
116              Ramsgate      0               NaN                  NaN   
117  Rivers Hull & Humber      0               NaN                  NaN   
118     Tees & Hartlepool      0               NaN                  NaN   

    UK Country  id_Port  
0     Scotland        0  
1          NaN        1  
2          NaN        2  
3          NaN        3  
4          NaN        4  
..         ...      ...  
114    England      114  
115    England      115  
116    England      116  
117    England      117  
118    England      118  

[119 rows x 6 columns]

Se ha creado la tabla/dataframe "UK_PORTS" con el identificador (PK) id_Port, detrayendo de los dataframes originales las columnas que caracterizan a los puertos.
Ahora, de esos primeros dataframes, una vez extraidas las columnas de atributos de puerto, se sustituye el nombre de puerto por el numero de identificador (que hace las veces de foreign key para la tabla "UK PORTS")

In [112]:
key_value_dict = dict(zip(df_ports_uk['Port'], df_ports_uk['id_Port'])) 

In [114]:
key_value_dict

{'Aberdeen': 0,
 'Appledore': 1,
 'Ardrishaig': 2,
 'Ayr': 3,
 'Barrow': 4,
 'Barry': 5,
 'Belfast': 6,
 'Berwick-On-Tweed': 7,
 'Bideford': 8,
 'Blyth': 9,
 'Boston': 10,
 'Bridgwater': 11,
 'Brightlingsea': 12,
 'Bristol': 13,
 'Buckie': 14,
 'Burry Port': 15,
 'Cairnryan': 16,
 'Cardiff': 17,
 'Chichester': 18,
 'Clyde': 19,
 'Coleraine': 20,
 'Corpach': 21,
 'Cowes, Isle of Wight ': 22,
 'Cromarty Firth': 23,
 'Dover': 24,
 'Dundee': 25,
 'Falmouth': 26,
 'Felixstowe': 27,
 'Fishguard': 28,
 'Fleetwood': 29,
 'Forth': 30,
 'Fowey': 31,
 'Fraserburgh': 32,
 'Garston': 33,
 "Gill's Bay Scotland": 34,
 'Glensanda': 35,
 'Goole': 36,
 'Great Yarmouth': 37,
 'Grimsby & Immingham': 38,
 'Harwich': 39,
 'Heysham': 40,
 'Holyhead': 41,
 "Hughtown (St Mary's)": 42,
 'Hull': 43,
 'Inverkeithing': 44,
 'Inverness': 45,
 'Ipswich': 46,
 'Kilroot': 47,
 'Kilroot Power Station Jetty': 48,
 "King's Lynn": 49,
 'Lancaster': 50,
 'Larne': 51,
 'Lerwick': 52,
 'Littlehampton': 53,
 'Liverpool': 54,


In [115]:
uk_freight = uk_ports_freight.iloc[:, [1] + list(range(3, len(uk_ports_freight.columns)-1))]
uk_freight.rename(columns={'FK_Port':'Port'}, inplace=True)

In [120]:
uk_freight

FK_Port       2001       2002       2003       2004  \
0                        Boston    847.184    765.628   1034.870    704.946   
1                       Bristol  10895.045  10083.423  11438.876  10759.172   
2                         Dover  19073.725  20211.622  18795.663  20752.669   
3                    Felixstowe  28353.531  25119.274  22282.379  23413.458   
4                    Fleetwood    1607.977   1520.991   1624.187   1661.891   
5                         Fowey   1535.342   1453.187   1447.116   1330.411   
6                         Goole   2633.452   2265.137   1913.460   2173.881   
7                Great Yarmouth    666.069    710.770    778.405    607.098   
8           Grimsby & Immingham  54830.711  55731.288  55931.016  57615.871   
9                       Harwich   2622.573   3495.334   4330.417   4264.181   
10                      Heysham   3823.501   3705.388   4082.962   3539.386   
11                         Hull  10585.613  10297.979  10528.802  12443.284   
12                     Ipswich    2923.963   3335.725   3888.275   3557.041   
13                    Liverpool  30288.314  30412.703  31684.319  32232.889   
14                      London   50654.008  51185.087  51027.682  53289.400   
15                   Manchester   7879.260   6279.353   6088.018   6634.166   
16                       Medway  14852.590  14839.617  15618.891  14534.977   
17                     Newhaven    997.600    862.622    949.411    929.069   
18                     Plymouth   1877.374   1854.494   2053.149   2166.785   
19                        Poole   1819.141   1798.352   1640.354   1754.492   
20                   Portsmouth   4282.034   4365.039   4221.697   4939.596   
21                     Ramsgate   1431.580   1848.206   1788.784   1701.805   
22                  River Trent   2395.789   2345.647   2309.360   2328.844   
23         Rivers Hull & Humber   7845.541   8901.588  10024.564   9241.697   
24                     Shoreham   1803.946   1785.966   1725.493   1686.253   
25                 Southampton   35689.032  34156.012  35772.610  38431.194   
26                   Sunderland   1021.455    928.234   1019.920   1117.064   
27            Tees & Hartlepool  50841.608  50447.469  53842.332  53819.227   
28                         Tyne   2468.886   2655.588   2763.470   2973.018   
29                      Belfast  13402.468  12824.817  13201.409  13558.563   
30  Kilroot Power Station Jetty   1264.288   1053.604   1041.428   1114.398   
31                        Larne   3519.739   4294.823   4319.454   4983.561   
32                  Londonderry   1059.926   1064.775   1172.044   1391.790   
33                  Warrenpoint   1480.397   1825.552   1879.509   1966.643   
34                     Aberdeen   3845.418   3645.040   3232.778   3888.060   
35                    Cairnryan   2014.228   2099.093   2327.709   2848.869   
36                        Clyde  11068.826   9733.094   9214.234  11507.189   
37               Cromarty Firth   2144.692   2658.438   3500.865   3207.856   
38                       Dundee   1100.906   1102.858   1016.166   1057.780   
39                        Forth  41607.233  42201.756  38751.792  34892.053   
40                    Glensanda   5470.854   5845.586   5322.386   5189.250   
41                   Loch Ryan       0.000      0.000      0.000      0.000   
42                       Orkney  18406.739  18812.232  14421.967  17933.829   
43                    Peterhead   1339.258   1342.705   1050.515    675.904   
44                   Stranraer    1404.379   1273.199   1273.566   1277.165   
45                   Sullom Voe  31165.651  29375.706  26360.270  23939.420   
46                      Cardiff   2739.000   2208.647   2287.000   2504.000   
47                    Fishguard    341.396    408.242    473.862    521.846   
48                     Holyhead   3228.607   3288.271   3328.712   3945.425   
49                Milford Haven  33791.592  34543.239  32737.114  38451.87

In [121]:
uk_freight['id_Port'] = uk_freight['FK_Port'].map(key_value_dict)

/tmp/ipykernel_16421/1414437250.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  uk_freight['id_Port'] = uk_freight['FK_Port'].map(key_value_dict)


Se ha incluido la columna id_Port que es la foreign key que la relaciona con Ports_UK:

In [122]:
uk_freight

FK_Port       2001       2002       2003       2004  \
0                        Boston    847.184    765.628   1034.870    704.946   
1                       Bristol  10895.045  10083.423  11438.876  10759.172   
2                         Dover  19073.725  20211.622  18795.663  20752.669   
3                    Felixstowe  28353.531  25119.274  22282.379  23413.458   
4                    Fleetwood    1607.977   1520.991   1624.187   1661.891   
5                         Fowey   1535.342   1453.187   1447.116   1330.411   
6                         Goole   2633.452   2265.137   1913.460   2173.881   
7                Great Yarmouth    666.069    710.770    778.405    607.098   
8           Grimsby & Immingham  54830.711  55731.288  55931.016  57615.871   
9                       Harwich   2622.573   3495.334   4330.417   4264.181   
10                      Heysham   3823.501   3705.388   4082.962   3539.386   
11                         Hull  10585.613  10297.979  10528.802  12443.284   
12                     Ipswich    2923.963   3335.725   3888.275   3557.041   
13                    Liverpool  30288.314  30412.703  31684.319  32232.889   
14                      London   50654.008  51185.087  51027.682  53289.400   
15                   Manchester   7879.260   6279.353   6088.018   6634.166   
16                       Medway  14852.590  14839.617  15618.891  14534.977   
17                     Newhaven    997.600    862.622    949.411    929.069   
18                     Plymouth   1877.374   1854.494   2053.149   2166.785   
19                        Poole   1819.141   1798.352   1640.354   1754.492   
20                   Portsmouth   4282.034   4365.039   4221.697   4939.596   
21                     Ramsgate   1431.580   1848.206   1788.784   1701.805   
22                  River Trent   2395.789   2345.647   2309.360   2328.844   
23         Rivers Hull & Humber   7845.541   8901.588  10024.564   9241.697   
24                     Shoreham   1803.946   1785.966   1725.493   1686.253   
25                 Southampton   35689.032  34156.012  35772.610  38431.194   
26                   Sunderland   1021.455    928.234   1019.920   1117.064   
27            Tees & Hartlepool  50841.608  50447.469  53842.332  53819.227   
28                         Tyne   2468.886   2655.588   2763.470   2973.018   
29                      Belfast  13402.468  12824.817  13201.409  13558.563   
30  Kilroot Power Station Jetty   1264.288   1053.604   1041.428   1114.398   
31                        Larne   3519.739   4294.823   4319.454   4983.561   
32                  Londonderry   1059.926   1064.775   1172.044   1391.790   
33                  Warrenpoint   1480.397   1825.552   1879.509   1966.643   
34                     Aberdeen   3845.418   3645.040   3232.778   3888.060   
35                    Cairnryan   2014.228   2099.093   2327.709   2848.869   
36                        Clyde  11068.826   9733.094   9214.234  11507.189   
37               Cromarty Firth   2144.692   2658.438   3500.865   3207.856   
38                       Dundee   1100.906   1102.858   1016.166   1057.780   
39                        Forth  41607.233  42201.756  38751.792  34892.053   
40                    Glensanda   5470.854   5845.586   5322.386   5189.250   
41                   Loch Ryan       0.000      0.000      0.000      0.000   
42                       Orkney  18406.739  18812.232  14421.967  17933.829   
43                    Peterhead   1339.258   1342.705   1050.515    675.904   
44                   Stranraer    1404.379   1273.199   1273.566   1277.165   
45                   Sullom Voe  31165.651  29375.706  26360.270  23939.420   
46                      Cardiff   2739.000   2208.647   2287.000   2504.000   
47                    Fishguard    341.396    408.242    473.862    521.846   
48                     Holyhead   3228.607   3288.271   3328.712   3945.425   
49                Milford Haven  33791.592  34543.239  32737.114  38451.87

Se hace lo mismo para UK_vessels:

In [123]:
uk_number_vessels_ports.columns

Index([             'Port', 'Major/Minor ports',        'Port group',
                      2009,                2010,                2011,
                      2012,                2013,                2014,
                      2015,                2016,                2017,
                  '2017.1',                2018,                2019,
                      2020,                2021],
      dtype='object')

In [124]:
uk_vessels_in_port = uk_number_vessels_ports.iloc[:, [0] + list(range(3, len(uk_ports_freight.columns)-1))]

In [127]:
uk_vessels_in_port['id_Port'] = uk_vessels_in_port['Port'].map(key_value_dict)

/tmp/ipykernel_16421/2666425809.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  uk_vessels_in_port['id_Port'] = uk_vessels_in_port['Port'].map(key_value_dict)


In [128]:
uk_vessels_in_port

Port  2009  2010  2011  2012  2013  2014  2015  2016  \
0               Aberdeen  1490  1525  1558  1536  1511  1488  1411  1344   
1              Appledore     0     0     0     0     0     1     1     0   
2             Ardrishaig    62    81   110    65    89    87    85    75   
3                    Ayr   166   244   160   125   173   148   134   120   
4                 Barrow    39    74    67    52    59    40    43    47   
..                   ...   ...   ...   ...   ...   ...   ...   ...   ...   
107  Weymouth & Portland   402   361   953   827   721   658   172   394   
108           Whitstable    22    44    40    30    30    27    25    30   
109                 Wick     0     1     0     0     0     7    20     5   
110              Wisbech    23    25    30    38    16    35    43    44   
111           Workington    60    50    91   132   146   152   143   123   

     2017  2017.1  2018  2019  2020  id_Port  
0    1251    1266  1298  1145   886        0  
1       0       0     0     0     0        1  
2      19      15    11    15     5        2  
3      84     165   197   247   237        3  
4      51      61    54    59    52        4  
..    ...     ...   ...   ...   ...      ...  
107   371     491   508   581   375      107  
108    36       2     0    14     4      108  
109     5      13    12    24    16      109  
110    57      63    90    96    59      110  
111   136     111   177   159   138      111  

[112 rows x 15 columns]

Por ejemplo, London solo sale en la tabla 'freight' pero no en 'vessels_in_port'. Es posible que haya que considerar entidades regionales que engloben distintos puertos (ver la columna 'Port group' de df_ports_uk)

Ya tenemos las tablas df_ports_uk, uk_freight y uk_vessels_in_port, que relacionan puertos, regiones, con numero de barcos y volumen de cargamento.\
Con la cantidad de tonelaje que entra en cada puerto (o sale) se puede identificar si el puerto es un puerto de carga (de containers), pesquero, deportivo o de otro tipo.



*Guardar .csv de los dataframes creados:

In [130]:
pwd()

'/home/gonzalopc/Documentos/programas_mios/TFM_AFI_2025/TFM_AFI_2025/AAA_code'

In [131]:
uk_vessels_in_port.to_csv('../home_uk/uk_vessels_in_port.csv', index=False)
uk_freight.to_csv('../home_uk/uk_freight.csv', index=False)
df_ports_uk.to_csv('../home_uk/uk_ports.csv', index=False)